engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/kafka/source/KafkaGenericNodeSourceFactory.scala

```diff
@@ -0,0 +1,72 @@
+package pl.touk.nussknacker.engine.kafka.source
+
+import pl.touk.nussknacker.engine.flink.api.timestampwatermark.TimestampWatermarkHandler
+import org.apache.flink.types.Nothing
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.NodeId
+import pl.touk.nussknacker.engine.api.context.{OutputVar, ProcessCompilationError, ValidationContext}
+import pl.touk.nussknacker.engine.api.context.transformation.{NodeDependencyValue, OutputVariableNameValue, SingleInputGenericNodeTransformation}
+import pl.touk.nussknacker.engine.api.definition.{DualParameterEditor, NodeDependency, Parameter, StringParameterEditor}
+import pl.touk.nussknacker.engine.api.editor.DualEditorMode
+import pl.touk.nussknacker.engine.api.process.ProcessObjectDependencies
+import pl.touk.nussknacker.engine.api.typed.typing.Typed
+import pl.touk.nussknacker.engine.flink.api.process.FlinkSource
+import pl.touk.nussknacker.engine.flink.util.context.{InitContextFunction, VariableProvider}
+import pl.touk.nussknacker.engine.kafka.{KafkaConfig, KafkaUtils, RecordFormatter}
+import pl.touk.nussknacker.engine.kafka.serialization.KafkaDeserializationSchemaFactory
+
+import scala.reflect.ClassTag
+
+
+class KafkaGenericNodeSourceFactory[T: ClassTag](deserializationSchemaFactory: KafkaDeserializationSchemaFactory[T],
```

Comment:
Can you write some documentation like in `KafkaSourceFactory`? Do you see some benefits of using `KafkaSourceFactory` instead of this? IMO we should:
- add deprecations in `KafkaSourceFactory`
- use this class in most places in our code where was used `KafkaSourceFactory` e.g. in generic model
- add migration guide that ppl should migrate to this approach
- add to release notes information that user now can use `#inputMeta`